loadtest/control/simplecontroller/controller.go

```diff
@@ -107,9 +111,38 @@ func (c *SimpleController) Run() {
 		return
 	}
 
+	var injectedAction *UserAction
+
 	cycleCount := 1 // keeps a track of how many times the entire cycle of actions have been completed.
+Loop:
 	for {
+		// run injected action (if any) first
+		if injectedAction != nil {
+			if resp := injectedAction.run(c.user); resp.Err != nil {
+				c.status <- c.newErrorStatus(resp.Err)
+			} else {
+				c.status <- c.newInfoStatus(resp.Info)
+			}
+			injectedAction = nil
+		}
+
+		// check for injected actions even if no scripted actions are present
+		select {
+		case ia := <-c.injectedActionChan:
+			injectedAction = ia
+			continue Loop
```

Comment:
Do we need the label here?

Reply:
If the code is changed in the future, such as another for loop added, this will ensure it does not break and explicitly indicates which loop the continue is targeting.

Reply:
IMO, if there is a `break` or `continue` at any nesting level inside a labeled loop then the label should be used.

Reply:
Loop has been refactored and no label is needed.